synthesizer/program/src/resources/credits.aleo

```diff
@@ -106,573 +149,496 @@ record credits:
 
 /**********************************************************************************************************************/
 
-// This function allows any staker to bond their microcredits to a validator and specify a withdrawal address.
-// The corresponding functions for 'bond_public' are 'unbond_public' and 'claim_unbond_public'.
-function bond_public:
-    // Input the validator's address.
-    input r0 as address.public;
-    // Input the withdrawal address.
-    input r1 as address.public;
-    // Input the amount of microcredits to bond.
-    input r2 as u64.public;
+// bond_validator is used to initiall bond a validator into the active committee.
+// bond_validator can also be used to increase the self-bond of the validator.
+function bond_validator:
+    input r0 as address.public;  // Withdrawal address.
+    input r1 as u64.public;      // Amount of microcredits to bond.
+    input r2 as u8.public;       // Commission rate.
 
-    // Determine if the amount is at least one credit.
-    gte r2 1_000_000u64 into r3;
-    // Enforce the amount is at least one credit.
+    // Determine if the amount is at least 1 credits.
+    gte r1 1_000_000u64 into r3;
+    // Enforce the amount is at least 1 credits.
     assert.eq r3 true;
 
-    // Determine if the caller is attempting to bond as a validator.
-    is.eq self.caller r0 into r4;
-    // Determine if the caller is a user account.
-    is.eq self.caller self.signer into r5;
-    // If the caller is attempting to bond as a validator, enforce the caller is a user account.
-    ternary r4 r5 true into r6;
-    assert.eq r6 true;
+    // Commission must be a percentage in the range 0-100 inclusive.
+    gt r2 100u8 into r4;
+    assert.neq r4 true;
 
```

Comment:
Should we force validator to have a separate withdrawal address?
```
// Prevent using hot-on-box validator address from withdrawing self-bond and rewards.
assert.neq self.signer r0;
```

Reply:
This seems like a good idea, will likely prevent future hacks.

Reply:
It's definitely best practice to make them separate. This does require that another key be kept secure and accessible, but validators should have strong processes around this already.